src/moscot/solvers/_output.py

```diff
@@ -167,6 +177,79 @@ def chain(self, outputs: Iterable["BaseSolverOutput"], scale_by_marginals: bool
+
+        Parameters
+        ----------
+        mode
+            Which threshold to use for sparsification. Can be one of:
+
+                - "threshold" - threshold below which entries are set to 0.0.
+                - "percentile" - determine threshold by percentile below which entries are set to 0. Hence, between 0
+                  and 100.
+                - "min_1" - choose the threshold such that each row has at least one non-zero entry.
+
+        threshold
+            Threshold or percentile depending on `mode`. If `mode` is `min_1`, `threshold` can be neglected.
+        batch_size
+            How many rows of the transport matrix to sparsify per batch.
+        n_samples
+            If `mode` is `percentile`, determine the number of samples based on which the percentile
+            is computed stochastically. Note this means that a matrix of shape `[n_samples, transport_matrix.shape[1]]`
+            has to be instantiated. If `None`, `n_samples` is set to batch_size.
+
+        Returns
+        -------
+        Nothing, but adds the sparsified transport matrix (:class:`scipy.sparse.csr_matrix`) to `self.sparsified_tmap`.
+
+        Note
+        ----
+        This function only serves for interfacing software which has to instantiate the transport matrix. Within moscot,
+        there is no point in using this function. We encourage users not to use this function unless it is necessary.
+        """
+        mode = SparsifyMode(mode)
+        if mode == SparsifyMode.THRESHOLD:
+            thr = threshold
+        elif mode == SparsifyMode.PERCENTILE:
+            n_samples = n_samples if n_samples is not None else batch_size
+            x = np.eye(self.shape[1], max(n_samples, self.shape[1]))
+            res = self.pull(x, scale_by_marginals=False)  # tmap @ indicator_vectors
+            thr = np.percentile(res, threshold)
+        elif mode == SparsifyMode.MIN_1:
+            thr = np.inf
+            for batch in range(0, self.shape[1], batch_size):
+                x = np.eye(self.shape[1], min(batch_size, self.shape[1] - batch), -(min(batch, self.shape[1])))
+                res = self.pull(x, scale_by_marginals=False)  # tmap @ indicator_vectors
+                thr_batch = res.max(axis=1).min()
+                thr = thr_batch if thr_batch < thr else thr
+            thr -= 1e-12  # necessary for the edge case
+        else:
+            raise NotImplementedError(mode)
+
+        tmaps_sparse: List[csr_matrix] = []
+        for batch in range(0, self.shape[1], batch_size):
+            x = np.eye(self.shape[1], min(batch_size, self.shape[1] - batch), -(min(batch, self.shape[1])))
+            res = self.pull(x, scale_by_marginals=False)  # tmap @ indicator_vectors
+            res[res <= thr] = 0
+            tmaps_sparse.append(csr_matrix(res))
+        self._sparsified_tmap = sparse_hstack(tmaps_sparse)
+
+    @property
+    def sparsified_tmap(self) -> Optional[csr_matrix]:
```

Comment:
From API standpoint: do you really want to save the sparse matrix in the `output`?
It's not being used in any of the downstream functions like BaseOutput.{push,pull}...
As an alternative, `compute_sparsification` could just return the sparse matrix or even return `MatrixSolverOutput` instead so that user can use `push/pull`, etc. For the latter feature, #492 is needed, aas `MatrixSolverOutput` is no longer abstract.

Reply:
I see your point and was thinking about the same. 
At the moment I save it as an attribute for the user to have more control while making it easily accessible by CellRank.

I.e., if I want to use the CR kernel, I can first create sparsification and then CellRank reads the sparsified transport matrix from the moscot class. If we only call sparsification after instantiating the moscot-CR kernel, the user cannot check the sparsified transport matrix before construction of the CR kernel.

I.e., we want CR to only read out results from moscot, rather than calling functions within moscot.

But open for discussions here

Reply:
I don't think saving it as `csr` makes sense, since within `moscot`, you're not really using this (as explained in the `Returns` section).
I'd prefer this function to be more useful than from an interfacing standpoint to CR and would prefer to return `MatrixSolverOutput` rather than setting an attribute which is public and most importantly, not used anywhere, as then we/the user can easily quantify what effect it has on push/pull, etc.
From the interface standpoint, since it's no longer saved, we could pass kwargs to the `MongeKernel` that sparsifies this as and inside use `{k: v.sparsify(**kwargs) for k, prob in prob.solution.items()}`

Reply:
okay, let's do it this way.